cibuildwheel/windows.py

```diff
@@ -88,71 +88,63 @@ def install_cpython(version: str, arch: str, nuget: Path) -> Path:
     return installation_path
 
 
-def install_pypy(version: str, arch: str, url: str) -> Path:
+def install_pypy(tmp: Path, arch: str, url: str) -> Path:
     assert arch == "64" and "win64" in url
     # Inside the PyPy zip file is a directory with the same name
     zip_filename = url.rsplit("/", 1)[-1]
     extension = ".zip"
     assert zip_filename.endswith(extension)
-    installation_path = CIBW_INSTALL_PATH / zip_filename[: -len(extension)]
+    installation_path = CIBW_CACHE_PATH / zip_filename[: -len(extension)]
     if not installation_path.exists():
-        pypy_zip = CIBW_INSTALL_PATH / zip_filename
+        pypy_zip = tmp / zip_filename
         download(url, pypy_zip)
         # Extract to the parent directory because the zip file still contains a directory
         extract_zip(pypy_zip, installation_path.parent)
     return installation_path
```

Comment:
Ahh, I see, the Lock is held globally before this function is called. What do you think about moving the lock inside this and the install_cpython function? Otherwise it's difficult to spot that it's locked. Also, I think that we'd get better concurrency allowing multiple versions to be installed simultaneously. What do you think?

Reply:
I don't think we'd really get better concurrency (given the current state of things):
- local => things will be cached if you really want to run cibuildwheel for many projects in parallel so the time the lock is being held is negligible.
- GHA starts by a sample build => sequential run.
- we might save a bit of time on other tests but I really don't think so.

That being said, it probably makes sense to do this for readability & overall design as you mentioned.